code/modules/mining/ores_coins.dm

```diff
@@ -23,9 +23,13 @@
 	..()
 
 /obj/item/weapon/ore/Crossed(atom/movable/AM)
+	var/show_message = TRUE
+	for(var/obj/item/weapon/ore/O in loc)

```

Comment:
I feel like this is going to make things even worst since it is adding an extra loop. It will be quite bad in tiles stacked with ores(not so uncommon).